test/e2e/parallel/registration_service_test.go

```diff
@@ -779,16 +776,52 @@ func assertGetSignupReturnsNotFound(t *testing.T, await wait.Awaitilities, beare
 	invokeEndpoint(t, "GET", route+"/api/v1/signup", bearerToken, "", http.StatusNotFound)
 }
 
+// waitForUserSignupReadyInRegistrationService waits and checks that the UserSignup is ready according to registration service /signup endpoint
+func waitForUserSignupReadyInRegistrationService(t *testing.T, registrationServiceURL, name, bearerToken string) map[string]interface{} {
```

Comment:
what having this utility func in https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/wait/host.go instead? 
Something like 
```
func (a *HostAwaitility) WaitForUserSignupReadyInRegistrationService(t *testing.T, name, bearerToken string)  map[string]interface{} {
  registrationServiceURL := a. RegistrationServiceURL
  ...
}
```

Reply:
well, the `invokeEndpoint` func is expecting a specific status, so maybe we should move that into the `wait` package and refactor all affected tests with the same logic as the one you're proposing in this PR?

Reply:
ok, thanks for the suggestion! I can improve the code structure, but I was thinking if it's better to address this refactoring in separate PR, mainly because:

1. we can check if this fixes the flaky test (or provides more insight about why is happening)
2. it will be less clear if I mix fixes with refactoring in same code change

WDYT?

Reply:
sounds good to me, let's address this refactoring in a subsequent PR

Reply:
ok, I'll work on it ASAP!